src/microbe_stage/editor/CellEditorComponent.cs

```diff
@@ -1455,14 +1455,16 @@ private bool IsUpgradingPossibleFor(OrganelleDefinition organelleDefinition)
                 }
                 else
                 {
-                    var replacedHex = editedMicrobeOrganelles.GetElementAt(hex);
                     var data = new OrganellePlacementActionData(organelle, hex, orientation)
                     {
                         CostMultiplier = CostMultiplier,
                     };
 
-                    if (replacedHex != null)
-                        data.ReplacedCytoplasm = new List<OrganelleTemplate> { replacedHex };
+                    var replacedHex = organelle.RotatedHexes.Select(h => editedMicrobeOrganelles.GetElementAt(hex + h))
+                        .WhereNotNull().ToList();
+
+                    if (replacedHex.Count > 0)
+                        data.ReplacedCytoplasm = replacedHex;
```

Comment:
```suggestion
                    var replacedHexes = organelle.RotatedHexes.Select(h => editedMicrobeOrganelles.GetElementAt(hex + h))
                        .WhereNotNull().ToList();

                    if (replacedHexes.Count > 0)
                        data.ReplacedCytoplasm = replacedHexes;
```

Suggestion to have grammatically more correct variable name.